### ISSUE TITLE
Adding new mergeStrat variant 

### ISSUE THREAD

**Issue by ErrorDAR32** [w0rthy/SortingVisualizer#6]
see https://github.com/ErrorDAR32/SortingVisualizer for the new variant
this variant that i called doubly (any similarity with any other merge sort in existence is purely coincidence) basically divides the merge space into 4 sublists with the help of 3 pointers those four being  (in order) the sorted list, the first mergeable list, the workspace, and the second mergeable list, and compares the first items of the last 3 to en up with a merged array.

I tried to comply as much as possible with your style and requirements to integrate the merge variant, but i have probably failed miserably in some pretty stupid way, i had to port the code from python and i dont know that much c++, i also dont know how to compile your project so i couldnt test it properly, if any error  can be signaled  for correction would be cool, and eventually merge the variant if you like it would be cooler!

**Comment by w0rthy** [comment]
Hey, do you have the original python version I can look at?

**Comment by ErrorDAR32** [comment]
Yes, it is right [here.](https://gist.github.com/ErrorDAR32/3c7a3d14833c6cc51ae2d02b26b7c392)
It was part of some sorting challenges i put to myself to make a better merge in place, i may or may not have been inspired by your videos :P
